torchrl/data/replay_buffers/storages.py

```diff
@@ -674,22 +683,32 @@ def load_state_dict(self, state_dict):
         self.initialized = state_dict["initialized"]
         self._len = state_dict["_len"]
 
-    @implement_for("torch", "2.3")
-    def _set_tree_map(self, cursor, data, storage):
+    _torch_ge_23 = implement_for("torch", "2.3")
+
+    def _set_tree_map_impl_ge_23(self, cursor, data, storage):
         def set_tensor(datum, store):
             store[cursor] = datum
 
         # this won't be available until v2.3
         tree_map(set_tensor, data, storage)
 
-    @implement_for("torch", "2.0", "2.3")
-    def _set_tree_map(self, cursor, data, storage):  # noqa: 534
+    _torch_20_to_23 = implement_for("torch", "2.0", "2.3")
+
+    def _set_tree_map_impl_20_to_23(self, cursor, data, storage):
         # flatten data and cursor
         data_flat = tree_flatten(data)[0]
         storage_flat = tree_flatten(storage)[0]
         for datum, store in zip(data_flat, storage_flat):
             store[cursor] = datum
 
+    def _set_tree_map(self, cursor, data, storage):
```

Comment:
I know this is not an elegant solution, so I'll try to figure out how to make `implement_for` work well with the compiler

Reply:
Oh I think I found a good way to do it, will push it up soon